### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Xento**

### CODE REVIEW

**Comment by Xento** [comment]
I hope you can merge it. I tried to merge your changes into my fork, but I couldn't handle it.
I'm to unexperienced with the git commandline.

**Comment by Xento** [comment]
I think now I got everything correct ;-)

File: devices/milightRF24.coffee [hunk 1]
```diff
@@ -78,7 +92,28 @@ module.exports = (env) ->
...
+        discoMode: discoMode,
+        brightness: brightness,
+        color:
+          r: r,
+          g: g,
+          b: b
+          
+      @.emit("ReceivedData", dataObj);
     
+  MilightRF24::__proto__ = events.EventEmitter.prototype
```
Comment (philip1986) [inline devices/milightRF24.coffee]: Why you do it that way?

Wouldn't `class MilightRF24 extends  events.EventEmitter` have the same effect?

Comment (Xento) [inline devices/milightRF24.coffee]: Maybe
This was how I found it in an example.

I will change it and try it out.

2015-08-04 14:43 GMT+02:00 philip1986 notifications@github.com:

**Comment by philip1986** [comment]
Could you please also add some commands to the code, specially how the classes `MilightRF24` and `MilightRF24Zone` are interconnected.

**Comment by Xento** [comment]
I added some comments and improved the code

**Comment by Xento** [comment]
Are this enough comments?

**Comment by philip1986** [comment]
Yes, thanks a lot. 
I will take some time to add some tests for it (I do it for all devices), because the complexity of the whole plugin makes it necessary.

**Comment by Xento** [comment]
When you will merge it?
Its running very smooth ;-)

**Comment by philip1986** [comment]
Sorry, its been a while. As I told you I like to have some tests for the plugin. So, I took your branch and add some basic ones. Would be nice if you extend then, e.g. when one device isn't in zone 0. 
I also did a change about the configuration and initialization of milightRF24, because I didn't like the approach to set the port via the device config.

I am going to close this PR and open a new one with my modifications. Please have a look at it.

Cheers, Philip

**Pull request closed without merging**